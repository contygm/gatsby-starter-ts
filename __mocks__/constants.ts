const glossaryDefinitions: Array<GlossaryElements> = [
    {
        html: `<h3>Definition</h3><p>Lemon drops pastry danish halvah. Lemon drops pastry danish halvah.</p>`,
        frontmatter: {
            title: 'Air',
            letter: 'A',
            date: '3-2-2021',
            type: 'glossary',
            syllables: 'a·ban·don',
            phonetics: 'əˈbandən',
            similarWords: ['/blog/test', '/wiki/test'],
            relatedPosts: ['/blog/test1', '/wiki/test1'],
            tags: ['two', 'three']
        }
    },
    {
        html: `<h3>Definition</h3><p>Lemon drops pastry danish halvah. Lemon drops pastry danish halvah.</p>`,
        frontmatter: {
            title: 'Atom',
            letter: 'A',
            date: '3-2-2021',
            type: 'glossary',
            syllables: 'a·ban·don',
            phonetics: 'əˈbandən',
            similarWords: ['/blog/test', '/wiki/test'],
            relatedPosts: ['/blog/test1', '/wiki/test1'],
            tags: ['two', 'one']
        }
    },
    {
        html: `<h3>Definition</h3><p>Lemon drops pastry danish halvah. Lemon drops pastry danish halvah.</p>`,
        frontmatter: {
            title: 'Cloud',
            letter: 'C',
            date: '3-2-2021',
            type: 'glossary',
            syllables: 'a·ban·don',
            phonetics: 'əˈbandən',
            similarWords: ['/blog/test', '/wiki/test'],
            relatedPosts: ['/blog/test1', '/wiki/test1'],
            tags: ['two', 'one']
        }
    },
    {
        html: `<h3>Definition</h3><p>Lemon drops pastry danish halvah. Lemon drops pastry danish halvah.</p>`,
        frontmatter: {
            title: 'Moon',
            letter: 'M',
            date: '3-2-2021',
            type: 'glossary',
            syllables: 'a·ban·don',
            phonetics: 'əˈbandən',
            similarWords: ['/blog/test', '/wiki/test'],
            relatedPosts: ['/blog/test1', '/wiki/test1'],
            tags: ['two']
        }
    },
    {
        html: `<h3>Definition</h3><p>Lemon drops pastry danish halvah. Lemon drops pastry danish halvah.</p>`,
        frontmatter: {
            title: 'Window',
            letter: 'W',
            date: '3-2-2021',
            type: 'glossary',
            syllables: 'a·ban·don',
            phonetics: 'əˈbandən',
            similarWords: ['/blog/test', '/wiki/test1'],
            relatedPosts: ['/wiki/test', '/blog/test1'],
            tags: ['three']
        }
    }
];

const featuredOrRelatedPosts: Array<IndexElements> = [
    {
        excerpt:
            'Lemon drops pastry danish halvah. Chupa chups fruitcake powder...',
        fields: {
            slug: '/baking-tutorial/'
        },
        frontmatter: {
            title: 'Baking Tutorial',
            date: '3-2-2021',
            description: 'a baking tutorial',
            tags: ['two', 'three'],
            headerImage: {
                childImageSharp: {
                    gatsbyImageData: {}
                }
            }
        }
    },
    {
        excerpt:
            'Cheesecake lemon drops Cheesecake lemon drops pastry danish halvah...',
        fields: {
            slug: '/ten-best-things/'
        },
        frontmatter: {
            title: '10 Best Things',
            description: 'a list of the ten best things',
            tags: ['one', 'two', 'three'],
            date: '2-2-2021',
            headerImage: {
                childImageSharp: {
                    gatsbyImageData: {}
                }
            }
        }
    },
    {
        excerpt: 'Chupa chups fruitcake powder dessert gingerbread gummies...',
        fields: {
            slug: '/guide/'
        },
        frontmatter: {
            title: 'Guide',
            description: 'a guide',
            tags: ['one', 'two'],
            date: '5-2-2021',
            headerImage: {
                childImageSharp: {
                    gatsbyImageData: {}
                }
            }
        }
    }
];

const previous = {
    fields: {
        slug: '/post-before'
    },
    frontmatter: {
        title: 'Before - A Post About Things'
    }
};

const next = {
    fields: {
        slug: '/post-after'
    },
    frontmatter: {
        title: 'After - A Post About Things'
    }
};

const siteData: SiteMetadata = {
    title: `Starter`,
    siteUrl: `https://www.domain.com`,
    author: {
        name: `Writer B. Author`,
        description:
            'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla quam velit, vulputate eu pharetra nec, mattis ac neque. Duis vulputate commodo lectus, ac blandit elit tincidunt id. Sed rhoncus, tortor sed eleifend tristique, tortor mauris molestie elit'
    },
    businessName: 'The Company',
    phone: '999-999-9999',
    email: 'email@email.com',
    description: 'a super duper site!',
    image: '/nothing.png'
};

export {
    featuredOrRelatedPosts,
    previous,
    next,
    siteData,
    glossaryDefinitions
};
