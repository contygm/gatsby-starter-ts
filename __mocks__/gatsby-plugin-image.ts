import React from 'react';
const plugin = jest.requireActual('gatsby-plugin-image');

const mockImage = ({ imgClassName, ...props } : {imgClassName: string }) =>
    React.createElement('img', {
        ...props,
        className: imgClassName
    });

module.exports = {
    ...plugin,
    GatsbyImage: jest.fn().mockImplementation(mockImage),
    StaticImage: jest.fn().mockImplementation(mockImage)
};
