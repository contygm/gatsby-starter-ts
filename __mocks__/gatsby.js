const React = require('react');
const gatsby = jest.requireActual('gatsby');

// NOTE: This mocks the graphql() function, Link component, and StaticQuery component.
// Allow testing components that use Link or GraphQL
module.exports = {
    ...gatsby,
    graphql: jest.fn(),
    Link: jest.fn().mockImplementation(
        // these props are invalid for an `a` tag
        ({
            activeClassName,
            activeStyle,
            getProps,
            innerRef,
            partiallyActive,
            ref,
            replace,
            to,
            ...rest
        }) =>
            React.createElement('a', {
                ...rest,
                href: to
            })
    ),
    StaticQuery: jest.fn(),
    useStaticQuery: jest.fn()
};
