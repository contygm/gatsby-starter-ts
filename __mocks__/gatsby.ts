import React from 'react';
const gatsby = jest.requireActual('gatsby');

// NOTE: This mocks the graphql() function, Link component, and StaticQuery component.
// Allow testing components that use Link or GraphQL
// https://stackoverflow.com/questions/54941990/how-to-strongly-type-jest-mocks
// https://stackoverflow.com/questions/48759035/mock-dependency-in-jest-with-typescript
module.exports = {
    ...gatsby,
    graphql: jest.fn(),
    Link: jest.fn().mockImplementation(
        // these props are invalid for an `a` tag
        ({
            activeClassName,
            activeStyle,
            getProps,
            innerRef,
            partiallyActive,
            ref,
            replace,
            to,
            ...rest
        }) =>
            React.createElement('a', {
                ...rest,
                href: to
            })
    ),
    StaticQuery: jest.fn(),
    useStaticQuery: jest.fn().mockImplementation(() => ({
        site: {
            siteMetadata: {
                title: `Starter`,
                siteUrl: `https://www.domain.com`,
                author: {
                    name: `Writer B. Author`,
                    description:
                        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla quam velit, vulputate eu pharetra nec, mattis ac neque. Duis vulputate commodo lectus, ac blandit elit tincidunt id. Sed rhoncus, tortor sed eleifend tristique, tortor mauris molestie elit'
                },
                businessName: 'The Company',
                phone: '999-999-9999',
                email: 'email@email.com',
                description: 'a super duper site!',
                image: '/nothing.png'
            }
        }
    }))
};
