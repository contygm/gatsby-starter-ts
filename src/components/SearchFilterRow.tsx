import { faCircleXmark } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import React, { useEffect, useState } from 'react';
import { TagsList } from './TagsList';

export const SearchFilterRow = (props: {
    tags: Array<{
        fieldValue: string;
        totalCount: number;
    }>;
    totalPostCount: number;
    activeTag: string;
    searchQuery: string;
    type: PostType;
    clearSearchQuery: () => void;
    handleFilterUpdate: (e: any) => void;
    handleSubmitSearch: (e: any) => void;
}) => {
    const [value, setValue] = useState(props.searchQuery);

    const handleChange = (event: any) => {
        const val = event.target.value;
        setValue(val);
    };

    useEffect(() => {
        setValue(props.searchQuery);
    }, [props.searchQuery]);

    const hasQuery = props.searchQuery !== "";

    return (
        <section className="section ">
            <div className="container is-max-desktop">
                <div className="columns">
                    <div className="column">
                        <form
                            className="field has-addons"
                            onSubmit={props.handleSubmitSearch}
                        >
                            <p className="control has-icons-right is-100-wide">
                                <input
                                    className="input is-100-wide"
                                    name="searchPost"
                                    data-testid="searchPost"
                                    type="search"
                                    placeholder="Find a post"
                                    value={value}
                                    key={props.searchQuery}
                                    onChange={handleChange}
                                    disabled={props.activeTag !== '' && props.activeTag !== 'all'}
                                />

                                {hasQuery && 
                                    <span>
                                        <FontAwesomeIcon 
                                        size="2xs" 
                                        className="icon is-small is-right search-clear-btn" 
                                        icon={faCircleXmark} 
                                        onClick={props.clearSearchQuery}
                                    />
                                    </span>
                                }
                                 
                            </p>
                            <p className="control is-100-wide">
                                <button
                                    className="button is-60-wide is-success"
                                    data-testid="searchPostSubmit"
                                    disabled={props.activeTag !== '' && props.activeTag !== 'all'}
                                >
                                    Search
                                </button>
                            </p>
                        </form>
                    </div>

                    <div className="column is-two-thirds m-auto">
                        <TagsList
                            totalPostCount={props.totalPostCount}
                            tags={props.tags}
                            activeTag={props.activeTag}
                            handleFilterUpdate={props.handleFilterUpdate}
                            type={props.type}
                            isSearchActive={hasQuery}
                        />
                    </div>
                </div>
            </div>
        </section>
    );
};
