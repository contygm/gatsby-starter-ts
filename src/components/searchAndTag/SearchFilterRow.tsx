import { faCircleXmark } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import React, { SyntheticEvent, useEffect, useState } from 'react';
import { TagsList } from './TagsList';

/**
 * search + filter rows props
 * @property {Array<{fieldValue: string, totalCount: number}>} tags - array of all tags with total tag count
 * @property {number} totalPostCount - total amount of posts
 * @property {string} activeTag - the current tag being used for filtering
 * @property {boolean} isSearchActive - true when the user is interacting with the search bar. When true, filter
 * will be disabled.
 * @property {PostType} type - the type of post being filtered/searched
 * @property {void} handleFilterUpdate - void function used as the action when filter is updated
 * @property {void} handleSubmitSearch - void function used as the action when search is submitted
 *
 * @memberof SearchFilterRow
 * @category Components
 */
interface SearchFilterRowProps {
    tags: {
        fieldValue: string;
        totalCount: number;
    }[];
    totalPostCount: number;
    activeTag: string;
    searchQuery: string;
    type: PostType;
    handleFilterUpdate: (e: SyntheticEvent) => void;
    handleSubmitSearch: (e: any) => void;
}
/**
 * A component with search and filter based functionalities. The search and
 * filter functionalities cannot be used at the same time. The component can be used with all
 * post types and is meant to be used on an index style page.
 *
 * `Filter:` The filter is based on the posts' tags. All tags used are display on the right and
 * the user can filter the displayed posts by clicking on the tag button.
 *
 * `Search:` The search has a clear button and will search when the user presses enter or clicks
 * the search button. The search returns results if the user's query is found in the post title,
 * or description
 *
 * **NOTE: The search functionality is currently not implemented**
 * @param {SearchFilterRowProps} props
 *
 * @see TagsList
 * @see IndexFilterWrapper
 * @class
 * @category Components
 */
export const SearchFilterRow = (props: SearchFilterRowProps) => {
    const [value, setValue] = useState(props.searchQuery);

    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        const val = event.target.value;
        setValue(val);
    };

    useEffect(() => {
        setValue(props.searchQuery);
    }, [props.searchQuery]);

    const hasQuery = props.searchQuery !== '';

    return (
        <section className="section">
            <div className="search-filter-wrapper">
                <div className="columns">
                    <div className="column">
                        <form
                            className="search-form"
                            onSubmit={props.handleSubmitSearch}
                        >
                            <p className="search-form-input-wrapper">
                                <input
                                    className="input"
                                    name="searchPost"
                                    data-testid="searchPost"
                                    type="search"
                                    placeholder="Find a post"
                                    value={value}
                                    key={props.searchQuery}
                                    onChange={handleChange}
                                    disabled={
                                        props.activeTag !== '' &&
                                        props.activeTag !== 'all'
                                    }
                                />
                            </p>
                            <p className="control">
                                <button
                                    className="primary-button"
                                    data-testid="searchPostSubmit"
                                    disabled={
                                        props.activeTag !== '' &&
                                        props.activeTag !== 'all'
                                    }
                                >
                                    Search
                                </button>
                            </p>
                        </form>
                    </div>

                    <div className="search-row-tags-wrapper">
                        <TagsList
                            totalPostCount={props.totalPostCount}
                            tags={props.tags}
                            activeTag={props.activeTag}
                            handleFilterUpdate={props.handleFilterUpdate}
                            type={props.type}
                            isSearchActive={hasQuery}
                        />
                    </div>
                </div>
            </div>
        </section>
    );
};
