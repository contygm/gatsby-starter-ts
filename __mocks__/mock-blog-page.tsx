/* istanbul ignore file */
import { BlogIndexProps } from '../src/pages/blog';
import { mockHeadProps, mockPageProps } from '../src/utils/test-props';
import { HeadProps, PageProps } from 'gatsby';
import { featuredOrRelatedPosts, siteData } from './constants';

const mockBlogs: BlogIndexProps = {
    site: siteData,
    allTags: {
        group: [
            { fieldValue: 'three', totalCount: 3 },
            { fieldValue: 'two', totalCount: 2 },
            { fieldValue: 'one', totalCount: 1 }
        ]
    },
    index: {
        totalCount: 3,
        nodes: featuredOrRelatedPosts
    },
    featured: {
        nodes: featuredOrRelatedPosts
    }
};

export const mockBlogPageData = mockPageProps(
    mockBlogs
) as PageProps<BlogIndexProps>;

export const mockBlogHeadData = mockHeadProps(
    mockBlogs
) as HeadProps<BlogIndexProps>;
