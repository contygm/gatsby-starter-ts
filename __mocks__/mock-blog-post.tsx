/* istanbul ignore file */
import { HeadProps, PageProps } from 'gatsby';
import { BlogPostProps } from '../src/templates/blog-post';
import { mockHeadProps, mockPageProps } from '../src/utils/test-props';
import { previous, next, featuredOrRelatedPosts, siteData } from './constants';

export const mockBlogPost: BlogPostProps = {
    site: {
        siteMetadata: siteData
    },
    markdownRemark: {
        id: 'slkj-dsjflks-dflkjl',
        excerpt:
            'Lemon drops pastry danish halvah. Chupa chups fruitcake powder...',
        html: `<h2>Header</h2><p>Lemon drops pastry danish halvah. Lemon drops pastry danish halvah. Lemon drops pastry danish halvah. Lemon drops pastry danish halvah.</p>`,
        tableOfContents: `<ul><li><p><a href="#liquorice-topping">Liquorice topping</a></p><ul><li><p><a href="#pastry-halvah-caramels">Pastry halvah caramels</a></p><ul> <li><a href="#wafer-oat-cake">Wafer oat cake</a></li></ul></li></ul></li></ul>`,
        frontmatter: {
            date: 'December 02, 2022',
            title: 'Baking Tutorial',
            description: 'a baking tutorial',
            related: ['/one-url/', '/two-url/', '/three-url/'],
            tags: ['one', 'two', 'three'],
            headerImage: {
                childImageSharp: {
                    gatsbyImageData: {}
                }
            }
        }
    },
    previous: previous,
    next: next,
    featured: {
        nodes: featuredOrRelatedPosts
    },
    related: {
        nodes: featuredOrRelatedPosts
    }
};

export const mockBlogPostData = mockPageProps(
    mockBlogPost
) as PageProps<BlogPostProps>;
export const mockBlogPostHeadData = mockHeadProps(
    mockBlogPost
) as HeadProps<BlogPostProps>;
