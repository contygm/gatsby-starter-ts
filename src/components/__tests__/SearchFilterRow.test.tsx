import React from 'react';
import { render } from '@testing-library/react';
import { SearchFilterRow } from '../SearchFilterRow';
import { mockBlogs } from '../../../__mocks__/mock-blog-page';

describe('SearchFilterRow', () => {
    it('renders correctly with a query', () => {
        const { asFragment } = render(<SearchFilterRow
			type={'blog'}
			tags={mockBlogs.allTags.group}
			activeTag={''}
			totalPostCount={mockBlogs.index.totalCount}
			clearSearchQuery={jest.fn()}
			handleFilterUpdate={jest.fn()}
			handleSubmitSearch={jest.fn()}
			searchQuery={'hello'}
		/>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders correctly without a query', () => {
        const { asFragment } = render(<SearchFilterRow
			type={'blog'}
			tags={mockBlogs.allTags.group}
			activeTag={''}
			totalPostCount={mockBlogs.index.totalCount}
			clearSearchQuery={jest.fn()}
			handleFilterUpdate={jest.fn()}
			handleSubmitSearch={jest.fn()}
			searchQuery={''}
		/>);
        expect(asFragment()).toMatchSnapshot();
    });

	it('renders correctly with an active tag', () => {
        const { asFragment } = render(<SearchFilterRow
			type={'blog'}
			tags={mockBlogs.allTags.group}
			activeTag={'one'}
			totalPostCount={mockBlogs.index.totalCount}
			clearSearchQuery={jest.fn()}
			handleFilterUpdate={jest.fn()}
			handleSubmitSearch={jest.fn()}
			searchQuery={''}
		/>);
        expect(asFragment()).toMatchSnapshot();
    });
});
