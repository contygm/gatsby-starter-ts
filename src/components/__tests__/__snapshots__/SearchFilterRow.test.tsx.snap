// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchFilterRow renders correctly with a query 1`] = `
<DocumentFragment>
  <section
    class="section "
  >
    <div
      class="container is-max-desktop"
    >
      <div
        class="columns"
      >
        <div
          class="column"
        >
          <form
            class="field has-addons"
          >
            <p
              class="control has-icons-right"
            >
              <input
                class="input"
                data-testid="searchPost"
                name="searchPost"
                placeholder="Find a post"
                type="search"
                value="hello"
              />
              <span>
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-circle-xmark fa-2xs icon is-small is-right search-clear-btn"
                  data-icon="circle-xmark"
                  data-prefix="fas"
                  data-testid="clearSearch"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M256 512c141.4 0 256-114.6 256-256S397.4 0 256 0S0 114.6 0 256S114.6 512 256 512zM175 175c9.4-9.4 24.6-9.4 33.9 0l47 47 47-47c9.4-9.4 24.6-9.4 33.9 0s9.4 24.6 0 33.9l-47 47 47 47c9.4 9.4 9.4 24.6 0 33.9s-24.6 9.4-33.9 0l-47-47-47 47c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9l47-47-47-47c-9.4-9.4-9.4-24.6 0-33.9z"
                    fill="currentColor"
                  />
                </svg>
              </span>
            </p>
            <p
              class="control"
            >
              <button
                class="button is-success"
                data-testid="searchPostSubmit"
              >
                Search
              </button>
            </p>
          </form>
        </div>
        <div
          class="column is-two-thirds m-auto"
        >
          <div
            class="field is-grouped is-grouped-multiline"
          >
            <div
              class="tags has-addons mx-2 tag-align show-pointer"
            >
              <div
                class="tag tag is-success"
                data-testid="all"
                id="all"
              >
                All
              </div>
              <span
                class="tag is-dark"
              >
                3
              </span>
            </div>
            <div
              class="tags has-addons mx-2 tag-align show-pointer"
            >
              <div
                class="tag "
                data-testid="three"
                id="three"
              >
                three
              </div>
              <span
                class="tag is-light"
              >
                3
              </span>
            </div>
            <div
              class="tags has-addons mx-2 tag-align show-pointer"
            >
              <div
                class="tag "
                data-testid="two"
                id="two"
              >
                two
              </div>
              <span
                class="tag is-light"
              >
                2
              </span>
            </div>
            <div
              class="tags has-addons mx-2 tag-align show-pointer"
            >
              <div
                class="tag "
                data-testid="one"
                id="one"
              >
                one
              </div>
              <span
                class="tag is-light"
              >
                1
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`SearchFilterRow renders correctly with an active tag 1`] = `
<DocumentFragment>
  <section
    class="section "
  >
    <div
      class="container is-max-desktop"
    >
      <div
        class="columns"
      >
        <div
          class="column"
        >
          <form
            class="field has-addons"
          >
            <p
              class="control has-icons-right"
            >
              <input
                class="input"
                data-testid="searchPost"
                disabled=""
                name="searchPost"
                placeholder="Find a post"
                type="search"
                value=""
              />
            </p>
            <p
              class="control"
            >
              <button
                class="button is-success"
                data-testid="searchPostSubmit"
                disabled=""
              >
                Search
              </button>
            </p>
          </form>
        </div>
        <div
          class="column is-two-thirds m-auto"
        >
          <div
            class="field is-grouped is-grouped-multiline"
          >
            <div
              class="tags has-addons mx-2 tag-align show-pointer"
            >
              <div
                class="tag tag is-success is-light"
                data-testid="all"
                id="all"
              >
                All
              </div>
              <span
                class="tag is-dark"
              >
                3
              </span>
            </div>
            <div
              class="tags has-addons mx-2 tag-align show-pointer"
            >
              <div
                class="tag is-success is-light"
                data-testid="three"
                id="three"
              >
                three
              </div>
              <span
                class="tag is-dark"
              >
                3
              </span>
            </div>
            <div
              class="tags has-addons mx-2 tag-align show-pointer"
            >
              <div
                class="tag is-success is-light"
                data-testid="two"
                id="two"
              >
                two
              </div>
              <span
                class="tag is-dark"
              >
                2
              </span>
            </div>
            <div
              class="tags has-addons mx-2 tag-align show-pointer"
            >
              <div
                class="tag is-success"
                data-testid="one"
                id="one"
              >
                one
              </div>
              <span
                class="tag is-dark"
              >
                1
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`SearchFilterRow renders correctly without a query 1`] = `
<DocumentFragment>
  <section
    class="section "
  >
    <div
      class="container is-max-desktop"
    >
      <div
        class="columns"
      >
        <div
          class="column"
        >
          <form
            class="field has-addons"
          >
            <p
              class="control has-icons-right"
            >
              <input
                class="input"
                data-testid="searchPost"
                name="searchPost"
                placeholder="Find a post"
                type="search"
                value=""
              />
            </p>
            <p
              class="control"
            >
              <button
                class="button is-success"
                data-testid="searchPostSubmit"
              >
                Search
              </button>
            </p>
          </form>
        </div>
        <div
          class="column is-two-thirds m-auto"
        >
          <div
            class="field is-grouped is-grouped-multiline"
          >
            <div
              class="tags has-addons mx-2 tag-align show-pointer"
            >
              <div
                class="tag tag is-success is-light"
                data-testid="all"
                id="all"
              >
                All
              </div>
              <span
                class="tag is-dark"
              >
                3
              </span>
            </div>
            <div
              class="tags has-addons mx-2 tag-align show-pointer"
            >
              <div
                class="tag is-success is-light"
                data-testid="three"
                id="three"
              >
                three
              </div>
              <span
                class="tag is-dark"
              >
                3
              </span>
            </div>
            <div
              class="tags has-addons mx-2 tag-align show-pointer"
            >
              <div
                class="tag is-success is-light"
                data-testid="two"
                id="two"
              >
                two
              </div>
              <span
                class="tag is-dark"
              >
                2
              </span>
            </div>
            <div
              class="tags has-addons mx-2 tag-align show-pointer"
            >
              <div
                class="tag is-success is-light"
                data-testid="one"
                id="one"
              >
                one
              </div>
              <span
                class="tag is-dark"
              >
                1
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
